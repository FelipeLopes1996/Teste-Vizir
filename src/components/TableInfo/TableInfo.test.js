import React from "react";
import { render, screen } from "@testing-library/react";
import { TableInfo } from ".";
describe("<TableInfo />", () => {
  it("should render title", () => {
    render(<TableInfo />);
    const element = screen.getByText(/Valores das chamadas/i);
    expect(element).toBeInTheDocument();
  });
  it("should render Th values", () => {
    render(<TableInfo />);
    const thOrigin = screen.getByText(/Origem/i);
    expect(thOrigin).toBeInTheDocument();
    const thDestiny = screen.getByText(/Destino/i);
    expect(thDestiny).toBeInTheDocument();

    const thMinutes = screen.getByText("$/min");
    expect(thMinutes).toBeInTheDocument();
  });
  it("should render Td values", () => {
    render(<TableInfo />);
    const tdEleven = screen.getAllByText(/011/i);
    expect(tdEleven).toHaveLength(6);
    const tdSixteen = screen.getAllByText(/016/i);
    expect(tdSixteen).toHaveLength(2);

    const tdSeventeen = screen.getAllByText(/017/i);
    expect(tdSeventeen).toHaveLength(2);

    const tdEighteen = screen.getAllByText(/018/i);
    expect(tdEighteen).toHaveLength(2);

    const tdOneDotNine = screen.getAllByText(/1.90/i);
    expect(tdOneDotNine).toHaveLength(2);
    const tdTwoDotNine = screen.getByText(/2.90/i);
    expect(tdTwoDotNine).toBeInTheDocument();
    const tdOneDotSeven = screen.getByText(/1.70/i);
    expect(tdOneDotSeven).toBeInTheDocument();
    const tdTwoDotSeven = screen.getByText(/2.70/i);
    expect(tdTwoDotSeven).toBeInTheDocument();
    const tdZeroDotSeven = screen.getByText(/0.90/i);
    expect(tdZeroDotSeven).toBeInTheDocument();
  });
  it("should render snapshot", () => {
    const { container } = render(<TableInfo />);
    expect(container).toMatchSnapshot();
  });
});
