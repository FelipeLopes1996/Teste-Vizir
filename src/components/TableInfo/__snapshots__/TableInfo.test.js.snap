// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableInfo /> should render snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL qyDuS"
  >
    <h1>
      Valores das chamadas
    </h1>
    <table
      class="sc-gsDKAQ iwoNMh"
    >
      <thead>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <th
            class="sc-hKwDye hKurbL"
          >
            Origem
          </th>
          <th
            class="sc-hKwDye hKurbL"
          >
            Destino
          </th>
          <th
            class="sc-hKwDye hKurbL"
          >
            $/min
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            011
          </td>
          <td>
            016
          </td>
          <td>
            1.90
          </td>
        </tr>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            016
          </td>
          <td>
            011
          </td>
          <td>
            2.90
          </td>
        </tr>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            011
          </td>
          <td>
            017
          </td>
          <td>
            1.70
          </td>
        </tr>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            017
          </td>
          <td>
            011
          </td>
          <td>
            2.70
          </td>
        </tr>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            011
          </td>
          <td>
            018
          </td>
          <td>
            0.90
          </td>
        </tr>
        <tr
          class="sc-dkPtRN cIDnkU"
        >
          <td>
            018
          </td>
          <td>
            011
          </td>
          <td>
            1.90
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
